src/render/texture_array_cache.rs

```diff
@@ -151,17 +151,16 @@ impl TextureArrayCache {
                 });
 
             for i in 0..count {
-                let columns = (atlas_size.x as f32 + spacing.x) / (tile_size.x + spacing.x);
-                let sprite_sheet_x: f32 = (i as f32 % columns).floor() * (tile_size.x + spacing.x);
-                let sprite_sheet_y: f32 = (i as f32 / columns).floor() * (tile_size.y + spacing.y);
+                let sprite_sheet_x: u32 = ((i % tile_count_x) as f32 * (tile_size.x + spacing.x)).min(atlas_size.x - tile_size.x) as u32;
```

Comment:
I need to calculate this out on paper by hand to check the math because this is the 3rd(?) time a change has been requested here, and I'm afraid we might just be swapping back and forth between two broken algorithms.

Reply:
It's still wrong as it doesn't take padding into account, but this should be correct for tilemaps with 0 padding. Previously the number of columns was a float and a float modulo was used, which was wrong as far as I can tell. The modulo first obtains the tile X, which can only be an unsigned integer, and whose formula is `index % number_of_elements_in_x_axis`. The number of elements in the X axis in this case is the number of columns. Then, for each tile, there is a stride of the tile width + spacing, as spacing only counts between tiles and not on the start nor end. Repeat for the Y axis.